### PR TITLE
Fix visibility for student comments in Instructor Student Records Page #5341 

### LINKED ISSUE #5341: Fix visibility for student comments in Instructor Student Records Page
In the Instructor Comments Page, when a visibility setting is checked or unchecked, other settings will be checked or unchecked automatically eg. checking "Can see giver's name" automatically checks "Can see your comment" as well.

This behaviour is not present in the Instructor Student Records Page however, and may lead to the DOM allowing combinations of visibility settings that are rejected (by not granting the conflicting visibility permissions) by the backend. As such, the Instructor Student Records Page will need to be changed to be consistent with the behaviour found in the Instructor Comments Page.

### PR DESCRIPTION
**Pull request by chowyb**
Fixes #5341

### CODE REVIEW

**Comment by chowyb** [comment]
Ready for review

**Comment by junhaoyap** [comment]
I have a question regarding this:

Does the Instructor Comments Page have a test suite to cover the functionality? Either way, if possible, try to design and create a test suite for this functionality in this page

File: src/main/webapp/js/instructorStudentRecords.js [hunk 1]
```diff
@@ -22,10 +22,21 @@ $(document).ready(function() {
...
 
-    $('input[type=checkbox]').click(function() {
+    $('input[type=checkbox]').click(function(e) {
         var table = $(this).parent().parent().parent().parent();
         var form = table.parent().parent().parent();
         var visibilityOptions = [];
+        var target = $(e.target);
+        
+        if (target.prop('class').includes('answerCheckbox') && !target.prop('checked')) {
+            target.parent().parent().find('input[class*=giverCheckbox]').prop('checked', false);
```
Comment (junhaoyap) [inline src/main/webapp/js/instructorStudentRecords.js]: I understand that you have copied this code from the other implementation, let's improve on it by moving away from all the ```.parent()```, namely, please look at this pull request here at https://github.com/TEAMMATES/teammates/pull/5389/files where @JosephineKwa has done a fine example of what I mean, perhaps you could sample that and look at how this whole part can be improved

**Comment by junhaoyap** [comment]
Please fix and ping me again via @ :)

**Comment by chowyb** [comment]
Fixed the issue with multiple parent calls, working on tests now.

On the topic of tests for Instructor Comments Page, this functionality (automatic check/uncheck of boxes) is not explicitly tested, as it only selects the checkboxes for `Can see your comment` within tests. This by chance covers a subset of all possible cases as a side effect of doing a HTML comparison of the main content.

**Comment by chowyb** [comment]
@junhaoyap ready for review :) if there's any tests I missed out on do let me know

File: src/main/webapp/js/instructorStudentRecords.js [hunk 2]
```diff
@@ -22,10 +22,22 @@ $(document).ready(function() {
 
     $('.panel-heading.student_feedback').click(toggleSingleCollapse);
 
-    $('input[type=checkbox]').click(function() {
+    $('input[type=checkbox]').click(function(e) {
```
Comment (junhaoyap) [inline src/main/webapp/js/instructorStudentRecords.js]: Could you also fix the 2 lines below this? Or else there will be conflicts with @JosephineKwa 's PR for sure

Comment (chowyb) [inline src/main/webapp/js/instructorStudentRecords.js]: Fixed using @JosephineKwa 's PR code

**Comment by junhaoyap** [comment]
This looks good now :) @damithc is it still the same procedure where I ping you to approve merges or do I ping someone else?

**Comment by damithc** [comment]
Still me @junhaoyap

**Pull request merged**